docs/writing-functions.md

```diff
@@ -208,12 +214,12 @@ defines functions, and can also include additional Ruby files defining classes
 and methods that assist in the function implementation.
 
 The "entrypoint" to the project, also called the "source", is a Ruby file. It
-can define any number of functions (with distinct names), although it is often
-good practice to create a separate Ruby file per function.
+can define any number of functions (with distinct names).
 
-By convention, the source file is often called `app.rb`, but you can give it
-any name. Projects can also have multiple source files that apply to different
-cases.
+By convention, the source file should generally be called `app.rb` and be
```

Comment:
I believe you should be able to use the `--source` with a relative file path to specify other files.

I haven't tried this out. In general I recommend using the default file path for samples.

Reply:
`gcloud functions deploy --source` lets you specify the root directory, but not an arbitrary source (as does `FUNCTION_SOURCE`). This is a point of confusion: the `gcloud` flag and the functions framework have different ideas of what "source" means. I'm dealing with this by pretending `gcloud`'s flag doesn't exist.